packages/frontend/core/src/modules/workbench/view/split-view/indicator.tsx

```diff
@@ -47,78 +35,86 @@ export const SplitViewMenuIndicator = memo(
+  isDragging?: boolean;
   menuItems?: React.ReactNode;
-  // import type { SyntheticListenerMap } from '@dnd-kit/core/dist/hooks/utilities' is not allowed
-  listeners?: any;
   setPressed?: (pressed: boolean) => void;
+  dragHandleRef?: React.RefObject<HTMLDivElement>;
 }
-export const SplitViewIndicator = ({
-  isDragging,
-  isActive,
-  menuItems,
-  listeners,
-  setPressed,
-}: SplitViewIndicatorProps) => {
-  const active = isActive || isDragging;
-  const [menuOpen, setMenuOpen] = useState(false);
+export const SplitViewIndicator = memo(
+  forwardRef<HTMLDivElement, SplitViewIndicatorProps>(
+    function SplitViewIndicator(
+      { isActive, menuItems, isDragging, dragHandleRef },
+      ref
+    ) {
+      const [menuOpen, setMenuOpen] = useState(false);
 
-  // prevent menu from opening when dragging
-  const setOpenMenuManually = useCallback((open: boolean) => {
-    if (open) return;
-    setMenuOpen(open);
-  }, []);
-  const openMenu = useCallback(() => {
-    setMenuOpen(true);
-  }, []);
+      // prevent menu from opening when dragging
+      const setOpenMenuManually = useCallback((open: boolean) => {
+        if (open) return;
+        setMenuOpen(open);
+      }, []);
 
-  const menuRootOptions = useMemo(
-    () =>
-      ({
-        open: menuOpen,
-        onOpenChange: setOpenMenuManually,
-      }) satisfies MenuProps['rootOptions'],
-    [menuOpen, setOpenMenuManually]
-  );
-  const menuContentOptions = useMemo(
-    () =>
-      ({
-        align: 'center',
-      }) satisfies MenuProps['contentOptions'],
-    []
-  );
+      const openMenu = useCallback(() => {
+        setMenuOpen(true);
+      }, []);
 
-  return (
-    <div data-is-dragging={isDragging} className={styles.indicatorWrapper}>
-      <Menu
-        contentOptions={menuContentOptions}
-        items={menuItems}
-        rootOptions={menuRootOptions}
-      >
-        <div className={styles.menuTrigger} />
-      </Menu>
-      <SplitViewMenuIndicator
-        open={menuOpen}
-        onOpenMenu={openMenu}
-        active={active}
-        setPressed={setPressed}
-        {...listeners}
-      />
-    </div>
-  );
-};
+      const menuRootOptions = useMemo(
+        () =>
+          ({
+            open: menuOpen,
+            onOpenChange: setOpenMenuManually,
+          }) satisfies MenuProps['rootOptions'],
+        [menuOpen, setOpenMenuManually]
+      );
+      const menuContentOptions = useMemo(
+        () =>
+          ({
+            align: 'center',
+          }) satisfies MenuProps['contentOptions'],
+        []
+      );
+
+      return (
+        <div
+          ref={ref}
+          data-is-dragging={isDragging}
+          className={styles.indicatorWrapper}
+        >
+          <Menu
+            contentOptions={menuContentOptions}
+            items={menuItems}
+            rootOptions={menuRootOptions}
+          >
+            <div className={styles.menuTrigger} />
+          </Menu>
+          <SplitViewMenuIndicator
```

Comment:
It seems that we don't need to split it into two components.

`SplitViewIndicator` and `SplitViewMenuIndicator `  should be one

Reply:
I think the intention was to make the code a bit cleaner. Let's keep it.